docs/protocols.ipynb

```diff
@@ -181,16 +145,300 @@
     "id": "b4d4bc702a5e"
    },
    "source": [
-    "## What is a Protocol\n",
+    "## What is a protocol? \n",
+    "\n",
+    "A protocol is a combination of the following two items: \n",
+    "- a `SupportsXYZ` class, which defines and documents all the magic functions that need to be implemented in order to support that given protocol \n",
+    "- the entrypoint function(s), which are exposed to the main cirq namespace as `cirq.xyz()`\n",
     "\n",
-    "TODO: describe the class SupportsXYZ, private methods, the entry-point cirq.xyz and the fallback strategies.\n",
+    "Note: While the protocol is technically both of these things, we refer to the public utility functions interchangeably as protocols. See the list of them below.\n",
     "\n",
     "\n",
     "## Cirq's protocols\n",
     "\n",
-    "TODO: draw a nice diagram of what areas the current protocols belong to and how they are interrelated\n",
+    "For a complete list of Cirq protocols, refer to the `cirq.protocols` package. \n",
+    "Here we provide a list of frequently used protocols for debugging, simulation and testing.\n",
+    "\n",
+    "\n",
+    "| Protocol | Description | \n",
+    "|----------|-------|\n",
+    "|`cirq.apply_channel`| High performance evolution under a channel evolution. |\n",
```

Comment:
what's the plan for keeping this up to date?

Reply:
Well...I did generate the full set and then trimmed it, so it could be generated very easily. 

```
for o in dir(cirq.protocols):
    obj = eval(f"cirq.protocols.{o}")
    if callable(obj) and not isinstance(obj, type):
        docs = obj.__doc__
        if docs is not None:
            first_line = docs.split('\n')[0]
            print(f"|`cirq.{o}`| {first_line} |")
        else:
            raise ValueError(f"{o} has no public docs and it's a protocol function!")
```

But I would think that the "notable" category for users is not going to change that much, and if we want to show case a new protocol we can include it here. 

Of course a better version would be to have an annotation `@showcase` or similar, that would then be fed to the generator... I'll open an issue for this, but I think for now this can go in this way. WDYT?

Reply:
https://github.com/quantumlib/Cirq/issues/3512

Reply:
Yes, it's probably true that they won't change enough to merit anything fancy. Just have to make sure it doesn't change *so* infrequently, we forgot that we documented and it becomes stale like 2+ years from now with us forgetting about it :)

Reply:
Agreed, added #3512